server/stream.go

```diff
@@ -237,6 +240,41 @@ type stream struct {
 
 	// Direct get subscription.
 	directSub *subscription
+
+	// dedup filter
+	msgDedupFilter msgDedupFilter
+}
+
+type msgDedupFilter struct {
+	msgDedupMap    map[string]*ddentry
+	msgDedupArr    []*ddentry
+	msgDedupTimer  *time.Timer
+	msgDedupIndex  int
+	msgDedupLoaded bool
+}
+
+const (
+	NoOfHashesForMsgDedup        = 20
+	NoOfBitsUsed          uint64 = 5000000000
```

Comment:
Is it the number of the bitmap? if yes so let's make it configurable

Reply:
I have a question - is there a bitmap per station or 1 per the entire system?
in case it is a map per station, we have to find a way we are not reaching memory leaks or memory overflow.

Regarding what you asked - the hash functions amount should be constant. In case we will give the user the option to configure the bitmap, it will be too complex for users who are not familiar with the bloom filter. It would be best if we could let the user choose the probability of duplications

Reply:
Yes, the bitmap is a map per station. It can cause memory leaks. Can you suggest a solution for this? Can we somehow prevent the user from creating too many stations with high memory requirement for bitmaps? 

Even if we keep the number of hash functions constant and take probability as user input, we need another parameter `N`(number of messages that reside in the broker) to derive `M`(number of bitmaps) and `M` is required to run the algorithm with memory optimisation.